bin/ch-checkns.c

```diff
@@ -106,41 +107,42 @@ int main(int argc, char *argv[])
    /* Claim /tmp for our namespace. You would think that because /tmp contains
       /tmp/newroot and it's a recursive bind mount, we could claim both in the
       same call. But, this causes pivot_root(2) to fail later with EBUSY. */
-   TRY (mount("/tmp", "/tmp", NULL, MS_REC | MS_BIND | MS_PRIVATE, NULL));
+   TRY (0 == mount("/tmp", "/tmp", NULL, MS_REC | MS_BIND | MS_PRIVATE, NULL));
 
    /* chdir to /tmp. This moves the process' special "." pointer to
       the soon-to-be root filesystem. Otherwise, it will keep pointing to the
       overmounted root. See the e-mail at the end of:
       https://git.busybox.net/busybox/tree/util-linux/switch_root.c?h=1_24_2 */
-   TRY (chdir("/tmp"));
+   TRY (0 == chdir("/tmp"));
 
    /* Move /tmp to /. (One could use this to directly enter the image,
       avoiding pivot_root(2) altogether. However, there are ways to remove all
       active references to the root filesystem. Then, the image could be
       unmounted, exposing the old root filesystem underneath. While
       Charliecloud does not claim a strong isolation boundary, we do want to
       make activating the UDSS irreversible.) */
-   TRY (mount("/tmp", "/", NULL, MS_MOVE, NULL));
+   TRY (0 == mount("/tmp", "/", NULL, MS_MOVE, NULL));
 
    /* Move the "/" special pointer to the new root filesystem, for the reasons
       above. (Similar reasoning applies for why we don't use chroot(2) to
       directly activate the UDSS.) */
-   TRY (chroot("."));
+   TRY (0 == chroot("."));
 
    /* Make a place for the old (intermediate) root filesystem to land. */
    if (mkdir("/newroot/oldroot", 0755) && errno != EEXIST)
       TRY (errno);
 
    /* Re-mount the image read-only. */
-   TRY (mount(NULL, "/newroot", NULL, MS_REMOUNT | MS_BIND | MS_RDONLY, NULL));
+   unsigned long flags = path_mount_flags("/newroot") | MS_REMOUNT | MS_BIND | MS_RDONLY;
```

Comment:
I prefer to put variable declarations at the beginning of blocks, because (1) then any variable is either in- or out-of-scope for the whole scope and (2) I can never remember if C makes the variable in scope for the whole block or just from declaration to the end.